packages/core/src/routes/interaction/verifications/mandatory-user-profile-validation.ts

```diff
@@ -23,7 +23,13 @@ const getMissingProfileBySignUpIdentifiers = ({
 }) => {
   const missingProfile = new Set<MissingProfile>();
 
-  if (signUp.password && !(user && isUserPasswordSet(user)) && !profile?.password) {
+  if (
+    signUp.password &&
+    !(user && isUserPasswordSet(user)) &&
+    !profile?.password &&
+    // Social identities can take place the role of password
+    !profile?.connectorId
+  ) {
```

Comment:
what if the user set username later?

Reply:
There is another verification logic in the `isUserPasswordSet` method. If social identity exists in the user data, then this method returns true.  Old logic from sijie.